### PR TITLE
fix(global header): export global header types 

### PR DESCRIPTION
**Pull request by saghdaey**
### Overview

Addressing [feedback](https://github.com/codecademy-engineering/Codecademy/pull/22024#discussion_r585939943) on my monolith PR to import the AnonHeaderVariant from GlobalHeader instead of declaring a new type "Variant" in the monolith's App Header

In the [PR i just merged](https://github.com/Codecademy/client-modules/pull/1451), I forgot to export the GlobalHeader/types file so came across an error when trying to use the new type in my monolith feature branch

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: EGG-536
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.3.3-alpha.603561.0<br/>@codecademy/styleguide@29.6.4-alpha.603561.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://603fe7e0f8d3ed09c9a709be--gamut-preview.netlify.app>https://603fe7e0f8d3ed09c9a709be--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/603fe7e0f8d3ed09c9a709be>Deploy Logs</a></p>

**Review by katiezutter — APPROVED** [review]

File: packages/gamut-labs/src/experimental/index.ts [hunk 1]
```diff
@@ -12,3 +12,4 @@ export * from './AppHeaderMobile/AppHeaderLinkMobile';
 export * from './AppHeaderMobile/AppHeaderMainMenuMobile';
 export * from './AppHeaderMobile/AppHeaderSubMenuMobile';
 export * from './GlobalHeader';
+export * from './GlobalHeader/types';
```
Comment (JoshuaKGoldberg) [inline packages/gamut-labs/src/experimental/index.ts:15]: Now that the types are being exported by the package itself, you might consider renaming `User` to be more specific. `HeaderUser`? `UserDetails`? `HeaderUserDetails`?

I'm not 100% sure if this is a good idea or what to name the thing if it is... 🤷

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
👍

File: packages/gamut-labs/src/experimental/index.ts (hunk 1 above)
Comment (saghdaey) [reply packages/gamut-labs/src/experimental/index.ts:15]: We're not using the other ones in the monolith...  I could do just` export { AnonHeaderVariant } from './GlobalHeader/types';` ? does that work instead?

Comment (JoshuaKGoldberg) [reply packages/gamut-labs/src/experimental/index.ts:15]: Sure!

Comment (saghdaey) [reply packages/gamut-labs/src/experimental/index.ts:15]: cool. ill push up that change! ty

Comment (saghdaey) [reply packages/gamut-labs/src/experimental/index.ts]: done! :)

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.3.3-alpha.a00c30.0<br/>@codecademy/styleguide@29.6.4-alpha.a00c30.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60401e0b7719930816876dc3--gamut-preview.netlify.app>https://60401e0b7719930816876dc3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60401e0b7719930816876dc3>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]
This too!

**Pull request merged**